OrbitGl/PresetsDataView.cpp

```diff
@@ -18,15 +18,57 @@
 #include "absl/strings/str_format.h"
 
 using orbit_client_protos::PresetFile;
+using orbit_client_protos::PresetInfo;
 
 PresetsDataView::PresetsDataView() : DataView(DataViewType::kPresets) {}
 
+std::string PresetsDataView::GetModulesList(const std::vector<ModuleView>& modules) const {
+  std::string modules_list;
+  for (const auto& module : modules) {
+    if (modules_list.empty()) {
+      absl::StrAppend(&modules_list, module.module_name);
+    } else {
+      absl::StrAppend(&modules_list, "\n", module.module_name);
+    }
+  }
+  return modules_list;
+}
+
+std::string PresetsDataView::GetFunctionCountList(const std::vector<ModuleView>& modules) const {
+  std::string function_count_list;
+  for (const auto& module : modules) {
+    if (function_count_list.empty()) {
+      absl::StrAppend(&function_count_list, module.function_count);
+    } else {
+      absl::StrAppend(&function_count_list, "\n", module.function_count);
+    }
+  }
+  return function_count_list;
+}
+
+std::string PresetsDataView::GetLoadStateString(
+    std::shared_ptr<orbit_client_protos::PresetFile> preset) const {
+  PresetLoadState load_state = GOrbitApp->GetPresetLoadState(preset);
+  switch (load_state) {
+    case kLoadable:
+      return "Yes";
+    case kPartiallyLoadable:
+      return "Partially";
+    case kNotLoadable:
+      return "No";
+    default:
+      return "";
+  }
```

Comment:
That's basically a `to_string` function for the enum defined in App.h. Ideally this is a free function. And there should be no `default`-case. That prevents the compiler from warning about missing enum values. A `return` after the switch is fine though.

Reply:
Moved this function to new PresetLoadState struct